src/RA_Core.cpp

```diff
@@ -1545,37 +1548,24 @@ void _WriteBufferToFile(const std::string& sFileName, const Document& doc)
     }
 }
 
-void _WriteBufferToFile(const std::string& sFileName, const DataStream& raw)
+void _WriteBufferToFile(const std::string& sFileName, const std::string& raw)
 {
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFileName.c_str(), "wb") == 0)
-    {
-        fwrite(raw.data(), 1, raw.size(), pf);
-        fclose(pf);
-    }
-}
 
-void _WriteBufferToFile(const std::string& sFileName, const std::string& sData)
-{
-    SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFileName.c_str(), "wb") == 0)
-    {
-        fwrite(sData.data(), 1, sData.length(), pf);
-        fclose(pf);
-    }
+    using FileH = std::unique_ptr<FILE, decltype(&std::fclose)>;
+    FileH myFile{ std::fopen(sFileName.c_str(), "wb"), std::fclose };
+    std::fwrite(static_cast<const void*>(raw.c_str()), sizeof(char), raw.length(), myFile.get());
```

Comment:
I checked that the files were still being written to by deleting the files first.